src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -264,6 +264,26 @@ private function addFragmentsSection(ArrayNodeDefinition $rootNode)
     private function addProfilerSection(ArrayNodeDefinition $rootNode)
     {
         $rootNode
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return 'file:' !== substr($v['dsn'], 0, 5); })
```

Comment:
you need to check whether the ``dsn`` key exists before accessing it, as you run it on unvalidated configuration.
The alternative is to put this trigger on the ``dsn`` node itself rather than on the ``profiler`` node, as it deals with a single value anyway